src/os/bluestore/bluefs_types.cc

```diff
@@ -51,6 +51,7 @@ void bluefs_super_t::decode(bufferlist::iterator& p)
   decode(block_size, p);
   decode(log_fnode, p);
   DECODE_FINISH(p);
+  log_fnode.recalc_allocated();
```

Comment:
i'd suggest update the DENC in `bluefs_fnode_t` instead so it will update its `allocated` at end of decode(). and also drop the `fnode.recalc_allocated()` call in  `BlueFS::_replay()` accordingly.

Reply:
:+1: